daemon/import.go

```diff
@@ -60,15 +62,24 @@ func (daemon *Daemon) ImportImage(src string, repository, tag string, msg string
 		rc = inConfig
 	} else {
 		inConfig.Close()
+		if len(strings.Split(src, "://")) == 1 {
+			src = "http://" + src
+		}
 		u, err := url.Parse(src)
 		if err != nil {
 			return err
 		}
-		if u.Scheme == "" {
-			u.Scheme = "http"
-			u.Host = src
-			u.Path = ""
+
+		// Checking the file referenced by the URL exists to avoid sending
+		// status code 200 to the client although the download process will fail
+		resp, err = http.Head(u.String())
+		if err != nil {
+			return err
+		}
+		if resp.StatusCode >= http.StatusBadRequest {
+			return fmt.Errorf("Unable to read file in %s: %s", u, resp.Status)
 		}
+
 		outStream.Write(sf.FormatStatus("", "Downloading from %s", u))
```

Comment:
As @LK4D4 reasoned, what if a tar file that its size is >500MB is being downloaded then it takes time till the "Downloading from..." message appears and that's sad.

I do see what you are trying to say here -- I guess both ways make sense to me.
If you insist on using `httputils.Download` over sending a HTTP head request, I will do that. 😄